### DISCUSSION TITLE
why don't have libunwind.so.1 after install?

### DISCUSSION THREAD

**Discussion by zxzfcsu** [libunwind/libunwind#665] [Q&A]
# My env:
- centos 7.6 with default glibc-2.17.
- I install glibc-2.35 in another directory /opt/glibc-2.35, and then changed glibc soft link like this
- - sln /opt/glibc-2.35/lib/libc.so /lib64/libc.so.6
- - sln /opt/glibc-2.35/bin/ld.so /usr/lib64/ld-linux-x86-64.so.2
- - sln /opt/glibc-2.35/lib/libm.so /usr/lib64/libm.so.6

# My installation of libunwind:
- download 1.5.0 tar
- echo "./configure --prefix=/opt/libunwind \n make \n make install \n make check" > install.sh
- yum -y install centos-release-scl
- yum -y install devtoolset-11 bison ca-certificates
- scl enable devtoolset-11 ./install.sh

# result
- only have libunwind.so.8.0.1 with libunwind.so and libunwind.so.8 link to this

# expect
libunwind.so.1

**Comment by zxzfcsu** [comment]
When I run 'ldd another.so', it indicates that 'another.so' needs 'libunwind.so.1'.

**Comment by bregma** [comment]
libunwind 1.5 builds a shared object with the SONAME of `libunwind.so.8`. If you have a binary that DT_NEEDS a shared object with a SONAME of `libunwind.so.1` then libunwind 1.5 will not satisfy it.

It is likely that your "another.so" was build against a completely different "libunwind", such as the one that's a part of LLVM. Alternatively, since Centos 7 was release in 2014, you may need an ancient version of libunwind released earlier than 1.5 (which was released in 2020). Try maybe libunwind 1.2 or something even more ancient from [nongnu.org](http://download.savannah.nongnu.org/releases/libunwind/).

This is not an issue with libunwind.

**Reply by zxzfcsu** [reply]
I'm not sure about that.
I logged into a pod, where another.so works well and the system version is Ubuntu 22.04.
It shows "libunwind-15:amd64: /usr/lib/x86_64-linux-gnu/libunwind.so.1" when exec "dpkg -S /usr/lib/x86_64-linux-gnu/libunwind.so.1".

**Reply by bregma** [reply]
The package [libunwind-15](https://packages.ubuntu.com/jammy/libunwind-15) on Ubuntu 22.14 is the LLVM libunwind. It's unfortunate that they chose to name their project with the same name as an existing established project that predates them by at least a decade.

This is not an issue with libunwind.

**Reply by zxzfcsu** [reply]
thank you!
